router/routes.go

```diff
@@ -136,6 +136,11 @@ func (rm *RouteManager) Add(route *Route) error {
 	}
 	route.closer = make(chan bool)
 	route.adapter = adapter
+	//Stop any existing route with this ID:
+	if( rm.routes[route.ID] != nil ){
```

Comment:
This question might be naive but should we throw an error instead of overwriting the route?